presto-main/src/main/java/com/facebook/presto/operator/DriverContext.java

```diff
@@ -159,24 +159,23 @@ public void recordBlocked(ListenableFuture<?> blocked)
 
     public void finished()
     {
+        freeMemory(memoryReservation.get());
+        freeSystemMemory(systemMemoryReservation.get());
```

Comment:
This PR does that actually. It's just not immediately clear (`finished()` is only called after freeing the operator memory). I'm making a minor refactoring to make that more obvious before merging this